app/src/androidTest/java/com/mapbox/maps/testapp/gestures/GestureActivityTest.kt

```diff
@@ -295,7 +300,83 @@ class GestureActivityTest : BaseMapTest() {
 
     rule.scenario.onActivity {
       it.runOnUiThread {
-        Assert.assertEquals(3.0, mapboxMap.cameraState.zoom, 0.0001)
+        assertEquals(3.0, mapboxMap.cameraState.zoom, 0.0001)
+      }
+    }
+  }
+
+  @Test
+  fun doubleTap_onScaleBegin_listener_test() {
+    val scaleListener = object : OnScaleListener {
+      override fun onScaleBegin(detector: StandardScaleGestureDetector) {
+        assertTrue(true)
+      }
+
+      override fun onScale(detector: StandardScaleGestureDetector) {
+      }
+
+      override fun onScaleEnd(detector: StandardScaleGestureDetector) {
+      }
+    }
+    rule.scenario.onActivity {
+      it.runOnUiThread {
+        mapView.gestures.addOnScaleListener(scaleListener)
+      }
+    }
+    onView(withId(R.id.mapView)).perform(ViewActions.doubleClick())
+    R.id.mapView.loopFor(1000)
+    mapView.gestures.removeOnScaleListener(scaleListener)
+  }
+
+  @Test
+  fun doubleTap_onScaleProgress_listener_test() {
+    val scaleListener = object : OnScaleListener {
+      override fun onScaleBegin(detector: StandardScaleGestureDetector) {
+      }
+
+      override fun onScale(detector: StandardScaleGestureDetector) {
+        assertTrue(true)
+      }
+
+      override fun onScaleEnd(detector: StandardScaleGestureDetector) {
+      }
+    }
+    rule.scenario.onActivity {
+      it.runOnUiThread {
+        mapView.gestures.addOnScaleListener(scaleListener)
+      }
+    }
+    onView(withId(R.id.mapView)).perform(ViewActions.doubleClick())
+    R.id.mapView.loopFor(1000)
+    mapView.gestures.removeOnScaleListener(scaleListener)
+  }
+
+  @Test
+  fun doubleTap_onScaleEnd_listener_test() {
```

Comment:
@ank27 all those 3 tests look pretty the same to me. IMHO we should have 3 tests:
1. double_tap_finished (quick zoom in regular)
2. double_tap_interrupted (quick zoom in interrupted by some other gesture)
3. two_finger_tap (quick zoom out)

All tests should have:
```
var onScaleBegin = false
var onScale = false
var onScaleEnd = false
val scaleListener = object : OnScaleListener {
      override fun onScaleBegin(detector: StandardScaleGestureDetector) {
          onScaleBegin = true
      }

      override fun onScale(detector: StandardScaleGestureDetector) {
        onScale = true
      }

      override fun onScaleEnd(detector: StandardScaleGestureDetector) {
         onScaleEnd = true
      }
    }
```
and check via `assertEquals` each flag after UI finished.

Reply:
updated with test cases.